src/builder.rs

```diff
@@ -90,49 +90,53 @@ impl Builder {
         }
     }
 
-    /// Set the I2C address to use. Defaults to 0x3C which is the most common address.
-    /// The other address specified in the datasheet is 0x3D. Ignored when using SPI interface.
-    pub fn with_i2c_addr(self, i2c_addr: u8) -> Self {
-        Self { i2c_addr, ..self }
-    }
-
     /// Set the rotation of the display to one of four values. Defaults to no rotation. Note that
     /// 90º and 270º rotations are not supported by
     /// [`TerminalMode`](../mode/terminal/struct.TerminalMode.html).
     pub fn with_rotation(self, rotation: DisplayRotation) -> Self {
         Self { rotation, ..self }
     }
 
-    /// Finish the builder and use I2C to communicate with the display
+    /// Finish the builder and use some interface communicate with the display
     ///
     /// This method consumes the builder and must come last in the method call chain
-    pub fn connect_i2c<I2C, CommE>(self, i2c: I2C) -> DisplayMode<RawMode<I2cInterface<I2C>>>
+    pub fn connect<I>(self, interface: I) -> DisplayMode<RawMode<I>>
     where
-        I2C: hal::blocking::i2c::Write<Error = CommE>,
+        I: WriteOnlyDataCommand<u8>,
     {
-        let properties = DisplayProperties::new(
-            I2cInterface::new(i2c, self.i2c_addr),
-            self.display_size,
-            self.rotation,
-        );
-        DisplayMode::<RawMode<I2cInterface<I2C>>>::new(properties)
+        let properties = DisplayProperties::new(interface, self.display_size, self.rotation);
+        DisplayMode::<RawMode<I>>::new(properties)
     }
+}
 
-    /// Finish the builder and use SPI to communicate with the display
+/// Builder struct for an I2C interface. Driver options and interface are set using its methods.
+#[derive(Clone, Copy)]
+pub struct I2CDIBuilder {
```

Comment:
Shouldn't this be called `I2cDiBuilder` to be compliant with the rust naming conventions? I think that `I2c` is also more common in HALs than `I2C`.

Reply:
Well, I2C is the real acronym, just like SPI. I don't really have any preference in how to call it but it's somewhat unfortunate to use an incorrectly spelled just for programming language naming convention sake.

If you prefer me to make that change, then no biggie.

Reply:
> Well, I2C is the real acronym, just like SPI. I don't really have any preference in how to call it but it's somewhat unfortunate to use an incorrectly spelled just for programming language naming convention sake.

IMO the reason to spell it incorrectly is increased readability, which was the primary reason I brought this up. `I2CDI` is harder to decipher than `I2cDi`. And this also makes this code more consistency with other crates in the embedded rust ecosystem, like the spelling of `Pwm` and `Qei` in `embedded_hal`.

For this reasons I would prefer to use `I2c` and `Spi` in this crate, but also in `display-interface`.

Reply:
Yeah, I'm definitely not going to change `display-interface`. I don't like the spelling of I2c or Spi at all (neither do I like Pwm and Qei, but they are what they are for whatever reason and are not going to change). If @jamwaffles would prefer `I2cDi` or maybe `I2CDi` I'd be happy to change it.

Reply:
I have opinions both ways lol. Both versions read strangely to me due to the pile of acronyms in the names. `display_interface_i2c` uses `I2CInterface` casing, so let's keep this PR as `I2CDIBuilder` for some semblance of consistency with that crate.

At the end of the day, these identifiers are going to be copypasta-ed and forgotten about so let's call this bikeshed resolved :).